File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        if not root:
            return 0
        def dfs(curr,path_max):
            if not curr:
                return 0
            if curr.val>=path_max:
                path_max = curr.val
                return dfs(curr.left,path_max)+dfs(curr.right,path_max)+1
            else:
                return dfs(curr.left,path_max)+dfs(curr.right,path_max)
        return dfs(root,root.val)
       

File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
 function dfs(curr:TreeNode|null,max_val:number):number{
    if (curr === null){
        return 0
    }
    if (curr.val>=max_val){
        return dfs(curr.left,curr.val)+dfs(curr.right,curr.val)+1
    }
    else{
        return dfs(curr.left,max_val)+dfs(curr.right,max_val)
    }
 }

function goodNodes(root: TreeNode | null): number {
    return dfs(root,root.val)
};